crates/onnx-ir/src/from_onnx.rs

```diff
@@ -349,15 +351,16 @@ pub fn parse_onnx(onnx_path: &Path) -> OnnxGraph {
 
     // Open the file
     let mut file = File::open(onnx_path).expect("Unable to open file");
-    let onnx_model: ModelProto =
+    let mut onnx_model: ModelProto =
         Message::parse_from_reader(&mut file).expect("Unable to parse ONNX file");
 
     // ONNX nodes must be topologically sorted per spec:
     // https://github.com/onnx/onnx/blob/main/docs/IR.md#graphs
-    debug_assert!(
-        onnx_model.graph.node.is_top_sorted(),
-        "Nodes are not topologically sorted"
-    );
+    //
+    // Some ONNX models are still not topographically sorted, so we'll just do it here
+    // to ensure that the model can be loaded.
+    onnx_model.graph = onnx_model.graph.map(|graph| topo_sort(graph));
```

Comment:
Instead of always sorting, maybe we should keep the current behavior and only sort if it is not already sorted.

Reply:
I'm thinking that it might be better to just always "sort" the graph, or really moreso to sanitize the input graph.
The `topo_sort` function (maybe a misnomer? I could rename it) does a lot of validation in the process of building the graph structure just to make sure it's valid overall before sending it through the rest of the codebase.

It'd probably save people some grief from seeing really odd errors downstream when the graph structure was wrong in the first place :)

Reply:
I'm not sure I've encountered invalid graphs a lot, especially since libraries that export to ONNX usually take care to follow the specs. But it can't hurt to check 🙂 

You're probably right that in the case that a user provides an invalid graph as input, this will definitely make it more clear as to what is wrong.